src/partition/repart.c

```diff
@@ -4775,6 +4775,28 @@ static int make_subvolumes_set(
         return 0;
 }
 
+static usec_t epoch_or_infinity(void) {
+        static usec_t cache;
+        static bool cached = false;
+        uint64_t epoch;
+        int r;
+
+        if (cached)
+                return cache;
+
+        r = secure_getenv_uint64("SOURCE_DATE_EPOCH", &epoch);
+        if (r >= 0) {
+                if (epoch <= UINT64_MAX / USEC_PER_SEC) { /* Overflow check */
+                        cached = true;
+                        return (cache = epoch * USEC_PER_SEC);
+                }
```

Comment:
doesn't matter much but i think we should probably debug log if the overflow condition.

a simple way would be to simply do `r = -ERANGE` here, and then change the `else if (r != -ENXIO)`  to just `if (r != -ENXIO)`